waku/v2/protocol/waku_filter/waku_filter.nim

```diff
@@ -48,16 +48,15 @@ proc notify*(filters: Filters, msg: WakuMessage, requestId: string = "") =
     # TODO: In case of no topics we should either trigger here for all messages,
     # or we should not allow such filter to exist in the first place.
     for contentFilter in filter.contentFilters:
-      if contentFilter.contentTopics.len > 0:
-        if msg.contentTopic in contentFilter.contentTopics:
+        if msg.contentTopic == contentFilter.contentTopic:
           filter.handler(msg)
           break
 
 proc unsubscribeFilters(subscribers: var seq[Subscriber], request: FilterRequest, peerId: PeerID) =
   # Flatten all unsubscribe topics into single seq
   var unsubscribeTopics: seq[ContentTopic]
   for cf in request.contentFilters:
-    unsubscribeTopics = unsubscribeTopics.concat(cf.contentTopics)
+    unsubscribeTopics = unsubscribeTopics.concat(@[cf.contentTopic])
```

Comment:
This was previously fairly convoluted logic to get all the nested `contentTopics` into a single `seq`. It should now be much simpler now that you have removed the redundance. You could likely replace all 3 lines by something like:
```
let unsubscribeTopics = request.contentFilters.mapIt(it.contentTopic)
```
More useful templates here: https://nim-lang.org/docs/sequtils.html#mapIt.t%2Ctyped%2Cuntyped

Reply:
Ahh thanks. Currently effecting the change